extension/app/src/lib/utils.ts

```diff
@@ -0,0 +1,62 @@
+export const randomString = (length: number): string => {
+  const charset =
+    "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz-._~";
+  let result = "";
+  const bytes = new Uint8Array(length);
+
+  crypto.getRandomValues(bytes);
+
+  for (let i = 0; i < length; i++) {
+    result += charset[bytes[i] % charset.length];
+  }
+
+  return result;
+};
+
+export const generatePKCE = async () => {
```

Comment:
ICYMI Auth0 provides all those utils already: https://auth0.com/docs/get-started/authentication-and-authorization-flow/authorization-code-flow-with-pkce/call-your-api-using-the-authorization-code-flow-with-pkce#create-code-challenge

Any reason not to use them?

Reply:
Yep, they were not working as is so @q helped me 😄